uw-frame-components/portal/notifications/services.js

```diff
@@ -4,68 +4,205 @@ define(['angular', 'jquery'], function(angular, $) {
 
   var app = angular.module('portal.notifications.services', []);
 
-  app.factory('notificationsService', ['$q','$http', 'miscService', 'PortalGroupService', 'keyValueService','SERVICE_LOC', 'KV_KEYS', function($q, $http, miscService, PortalGroupService, keyValueService, SERVICE_LOC, KV_KEYS) {
+  app.factory('notificationsService', ['$q','$http', '$log', '$filter', 'miscService', 'PortalGroupService', 'keyValueService','SERVICE_LOC', 'KV_KEYS', function($q, $http, $log, $filter, miscService, PortalGroupService, keyValueService, SERVICE_LOC, KV_KEYS) {
     // LOCAL VARIABLES
     var filteredNotificationPromise;
     var dismissedPromise;
-    var dismissedIds = [];
 
     /////////////////////
     // SERVICE METHODS //
     /////////////////////
+    
+    /**
+     * @typedef {Object} Notification
+     * @property {number} id
+     * @property {string} title
+     * @property {string} actionURL
+     * @property {boolean} dismissable
+     * @property {boolean} false
+     * @property {string} actionALt
+     * @property {string=} dataURL
+     * @property {string=} dataObject
+     * @property {Object=} dataArrayFilter
+     */
+    
+    /**
+     * @typedef {Object} NotificationReturnObject
+     * @property {Notification[]} dismissed
+     * @property {Notification[]} notDismissed
+     */
+    
     /**
      * Get all notifications at the given URL (as set in app-config.js or override.js)
-     * @returns {*} A promise which returns an object containing notifications arrays
+     * @returns {Promise<NotificationReturnObject>} A promise which returns an object containing notifications arrays
      */
     var getAllNotifications = function() {
-      return $q.all([$http.get(SERVICE_LOC.notificationsURL, {cache : true}), getDismissedNotificationIds()])
-          .then(function(result) {
-            return sortNotifications(result[0].data.notifications, result[1]);
-          },
-          function(reason) {
-            miscService.redirectUser(reason.status, 'notifications json feed call');
-          }
-        );
+      return $http.get(SERVICE_LOC.notificationsURL, {cache : true}).then(
+        function(result) {
+          return separateNotificationsByDismissal(result.data.notifications).then(
+            function(results){
+              return results;
+            },function(reason){
```

Comment:
Switch to catch block? Also, it might be wise to add brief comments throughout the chained promises so people can understand what's happening at a glance (i.e. what are "result" and "results" expected to be in this context?).